### PR TITLE
Add service layer

### LINKED ISSUE #275: Add service layer
##### **Overview of the feature request**

Give the possibility to generate service layer 

##### **Motivation for or Use Case**

Today all code is in controller 

##### **Related issues or PR**

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by ivanmonteiro**
Give the possibility to generate service layer

Fix #275

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by nicolas63** [comment]
Already in draft mode or you need review ?

**Comment by ivanmonteiro** [comment]
> Already in draft mode or you need review ?

I still have to solve some conflicts and fix. Hope to provide a review-able PR in a couple of days.

**Comment by ivanmonteiro** [comment]
> > Already in draft mode or you need review ?
> 
> I still have to solve some conflicts and fix. Hope to provide a review-able PR in a couple of days.

I think it is ready for review.
The latest commit (service files/namespace refactoring) added lots of changes (mostly namespace changes). 
If you think it is better to split this PR into smaller ones let me know.

**Comment by nicolas63** [comment]
Maybe we can rename Infrastructure folder in WebProject to Configuration ? This can be done in other PR

**Comment by nicolas63** [comment]
with new project they are nothing to add in dockerfile.ejs ?

File: generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    .ThenInclude(<%= relationship.joinEntityNameCamelCased %> => <%= relationship.joinEntityNameCamelCased %>.<%= relationship.otherEntityNamePascalized %>)
+                <%_ }
+            }); _%>
+                .SingleOrDefaultAsync(<%= camelCasedEntityClass %> => <%= camelCasedEntityClass %>.Id == id);
+            return result;
+        }
+
+        public async Task Delete(long id)
+        {
+            _log.LogDebug($"REST request to delete <%= pascalizedEntityClass %> : {id}");
```
[the remark this answers is not in the document]
Comment (ivanmonteiro) [reply generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs:106]: Should I remove log on service?

**Comment by ivanmonteiro** [comment]
> with new project there is nothing to add in dockerfile.ejs ?

I'll check the dockerfile, it probably needs some changes

> With jdl we can omit service creation with `service all with serviceImpl except Employee, Job`
> In your case the controller is good but the service is generated

I'll check what is going on.

File: generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicolas63) [reply generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs:106]: You have remove log on other method i think you can remove here too

**Comment by nicolas63** [comment]
> 
> 
> Maybe we can rename Infrastructure folder in WebProject to Configuration ? This can be done in other PR

I will do it

**Comment by ivanmonteiro** [comment]
> > Maybe we can rename Infrastructure folder in WebProject to Configuration ? This can be done in other PR
> 
Agree, it is better on other PR, after this PR is merged, because some files at that folder were moved.

> I will do it
Allright.

File: generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivanmonteiro) [reply generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs]: removed this service log with latest commit

**Comment by ivanmonteiro** [comment]
> with new project there is nothing to add in dockerfile.ejs ?

I just changed the dockerfile, now it restores the new ..csproj projects

```
# Restore dependencies of .net core projects taking advantage of docker layer caching
COPY src/*/*.csproj ./src/
RUN for file in $(ls src/*.csproj); do mkdir -p ${file%.*} && mv $file ${file%.*}; done
RUN dotnet restore "src/JhipsterSampleApplication/JhipsterSampleApplication.csproj"
```

File: generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivanmonteiro) [reply generators/entity-server/templates/dotnetcore/src/Project.Domain.Services/Service.cs.ejs]: also fixed the service generation

**Comment by ivanmonteiro** [comment]
> With jdl we can omit service creation with `service all with serviceImpl except Employee, Job`
> In your case the controller is good but the service is generated

Solved this with the latest commit

**Comment by nicolas63** [comment]
Everything looks good to me. It's a really nice job, thanks for your help on the project 😃

**Comment by ivanmonteiro** [comment]
Claimed bounty https://opencollective.com/generator-jhipster/expenses/23848

**Comment by pascalgrimaud** [comment]
@ivanmonteiro : just approved :)

**Pull request merged**